src/NuGet.Core/NuGet.Packaging/Rules/NoRefOrLibFolderInPackageRule.cs

```diff
@@ -0,0 +1,122 @@
+            {
+                //if you can't find the ref and lib folder, then find the build folder
+                if (buildFrameworks.Length != 0)
+                {
+                    //if you can find any folders other than native or any, raise an NU5127
+                    if (buildFrameworks.Any(t => (FrameworkConstants.DotNetAll.Satisfies(t) || IsValidFramework(t))
+                        && t.GetShortFolderName() != "dotnet"
+                        && t.GetShortFolderName() != "native"))
+                    {
+                        var possibleFrameworks = buildFrameworks.
+                            Where(t => t.IsSpecificFramework && t.GetShortFolderName() != "dotnet" && t.GetShortFolderName() != "native").
+                            Select(t => t.GetShortFolderName()).ToArray();
+
+                        (var tfmNames, var suggestedDirectories) = GenerateWarningString(possibleFrameworks);
+                        
+                        var issue = new List<PackagingLogMessage>();
+                        issue.Add(PackagingLogMessage.CreateWarning(string.Format(MessageFormat, tfmNames, suggestedDirectories),
+                            NuGetLogCode.NU5127));
+                        return issue;
+                    }
+                }
+            }
+
+            return Array.Empty<PackagingLogMessage>();
+        }
+
+        private (string, string) GenerateWarningString(string[] possibleFrameworks)
+        {
+            var tfmNames = new StringBuilder();
+            var suggestedDirectories = new StringBuilder();
+            if (possibleFrameworks.Length > 1)
+            {
+                for (int i = 0; i < possibleFrameworks.Length; i++)
+                {
+                    if (i != possibleFrameworks.Length - 1)
+                    {
+                        tfmNames.Append(possibleFrameworks[i] + ", ");
+                    }
+                    else
+                    {
+                        tfmNames.Append("and " + possibleFrameworks[i]);
+                    }
+
+                    tfmNames.AppendFormat("-lib/{0}/_._", possibleFrameworks[i]).AppendLine();
+                }
+            }
+            else
+            {
+                tfmNames.Append(possibleFrameworks[0]);
+                suggestedDirectories.AppendFormat("-lib/{0}/_._", possibleFrameworks[0]);
+            }
+
+            return (tfmNames.ToString(), suggestedDirectories.ToString());
+
+        }
+
+        private static IEnumerable<ContentItemGroup> GetContentForPattern(ContentItemCollection collection, PatternSet pattern)
+        {
+            return collection.FindItemGroups(pattern);
+        }
+
+        private static IEnumerable<NuGetFramework> GetGroupFrameworks(IEnumerable<ContentItemGroup> groups)
+        {
+            return groups.Select(e => ((NuGetFramework)e.Properties["tfm"]));
+        }
+
+        private static bool IsValidFramework(NuGetFramework framework)
```

Comment:
InvalidFrameworkFolderRule has a utility to achieve this alraedy. 
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/Rules/InvalidFrameworkFolderRule.cs

Consider refactoring that into a more reusable fashion.

Reply:
extracting the method into an internal static class within the same dir is probably a good idea.

Reply:
I don't think you addressed this. 

It's the same method in both this and the other rule correct? 
They can both call into an internal static utility.

Reply:
👋